### PR TITLE
Adding test for helper function

### PR DESCRIPTION
**Pull request by iRohitSingh**
@therewasaguy can you please! review this. I also tried to add the test for the _checkFileFormats but I am not able to debug it. Is there any way to put the breakpoint in test or console.log?

### CODE REVIEW

**Comment by iRohitSingh** [comment]
@therewasaguy can you please review this, I just wanted to know if I am on right track or not. sorry! to disturb you.

File: test/tests/p5.Helpers.js [hunk 1]
```diff
@@ -0,0 +1,16 @@
+var expect = chai.expect;
+
+describe('Testing helpers function', function () {
+  it('p5.prototype.freqToMidi helper function', function () {
+    let Midi = p5.prototype.freqToMidi(880);
```
Comment (therewasaguy) [inline test/tests/p5.Helpers.js:5]:
```suggestion
    const midi = p5.prototype.freqToMidi(880);
```

File: test/tests/p5.Helpers.js [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+    expect(Midi).to.equal(81);
+  });
+  it('p5.prototype.midiToFreq  helper function', function () {
+    let Freq = p5.prototype.midiToFreq(720);
```
Comment (therewasaguy) [inline test/tests/p5.Helpers.js:9]:
```suggestion
    const freq = p5.prototype.midiToFreq(720);
```

**Review by therewasaguy — CHANGES_REQUESTED** [review]
really helpful to have these tests, thank you!

Looks like this would work but the convention is to use lower case for variable names, and to use `const` instead of `let` if the variable won't be re-assigned within its scope.

**Comment by therewasaguy** [comment]
> I also tried to add the test for the _checkFileFormats but I am not able to debug it. Is there any way to put the breakpoint in test or console.log?

Great question!

When running the tests (by opening `http://localhost:8000/test` in one or multiple browsers to ensure compatibility across browsers) you should be able to use `console.log` statements.

And you can also set breakpoints in each browser. For example, [here's how to set a breakpoint in Chrome](https://developer.chrome.com/docs/devtools/javascript/breakpoints/
). After opening the JS console, I like to use the shortcut command-shift-p to search for the name of the file where I want to set the breakpoint. Even though the code runs on the compiled `lib/p5.sound.js` file, we can set breakpoints on files from the `/src` directory thanks to the `p5.sound.js.map` file that maps back to the source files.

**Comment by iRohitSingh** [comment]
@therewasaguy thankyou! started amending after your suggestion.

File: test/tests/p5.Helpers.js (hunk 2 above)
Comment (iRohitSingh) [reply test/tests/p5.Helpers.js]: Fixed

File: test/tests/p5.Helpers.js (hunk 1 above)
Comment (iRohitSingh) [reply test/tests/p5.Helpers.js]: Fixed

**Comment by iRohitSingh** [comment]
@therewasaguy Fixed everything :). I also added a test for the sound formats. is anything left in the helper function because I think other functions are not trivial to test?
waiting for your reply :)

File: test/tests/p5.Helpers.js [hunk 3]
```diff
@@ -0,0 +1,35 @@
+const expect = chai.expect;
+
+describe('Testing helpers function', function () {
+  it('p5.prototype.freqToMidi helper function', function () {
+    const midi = p5.prototype.freqToMidi(880);
+    expect(midi).to.equal(81);
+  });
+
+  it('p5.prototype.midiToFreq helper function', function () {
+    const freq = p5.prototype.midiToFreq(720);
```
Comment (therewasaguy) [inline test/tests/p5.Helpers.js:10]: 720 is outside of the typical MIDI (and frequency) range, which goes from 0 to 127, might be best to use a number within that range?

File: test/tests/p5.Helpers.js [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+    expect(freq).to.equal(261.6255653005986);
+  });
+
+  it('p5.prototype.soundFormats helper function', function () {
+    // setting file format so that if we don't provide extension
+    // our file will be loaded because _checkFileFormats add it for us.
+
+    p5.prototype.soundFormats('mp3');
+    const file = p5.prototype._checkFileFormats('a');
+    expect(file).to.be.equal('a.mp3');
```
Comment (therewasaguy) [inline test/tests/p5.Helpers.js:25]: This works, but ideally our test won't depend on a method with an underscore prefix (which, according to convention, is private and so could change or be deleted without us needing to update the API)

**Review by therewasaguy — COMMENTED** [review]
nice! Just a couple suggestions

File: test/tests/p5.Helpers.js (hunk 4 above)
Comment (therewasaguy) [reply test/tests/p5.Helpers.js:25]: can't think of a good way to do this off the top of my head...

File: test/tests/p5.Helpers.js (hunk 3 above)
Comment (iRohitSingh) [reply test/tests/p5.Helpers.js:10]: ok

File: test/tests/p5.Helpers.js (hunk 4 above)
Comment (iRohitSingh) [reply test/tests/p5.Helpers.js:25]: ok! I will try to come with some other function to test soundformat or I will leave it as it is.

File: test/tests/p5.Helpers.js (hunk 3 above)
Comment (iRohitSingh) [reply test/tests/p5.Helpers.js]: Done

**Comment by iRohitSingh** [comment]
@therewasaguy I searched throughout the codebase and didn't find a solution other than the check-file format. If you want I can remove it from the test.
can you please merge this I think it is ready?

File: test/tests/p5.Helpers.js (hunk 4 above)
Comment (iRohitSingh) [reply test/tests/p5.Helpers.js:25]: sorry! I am unable to find any better solution other than this.

**Review by therewasaguy — APPROVED** [review]

File: test/tests/p5.Helpers.js (hunk 4 above)
Comment (therewasaguy) [reply test/tests/p5.Helpers.js:25]: yeah, I'm not sure either but thanks for considering this feedback

**Comment by therewasaguy** [comment]
thank you @iRohitSingh !

**Comment by iRohitSingh** [comment]
@therewasaguy thank you. This week I will be inactive because I am creating a GSOC proposal for this project.

**Pull request merged**